zeebe/snapshot/src/test/java/io/camunda/zeebe/snapshots/testing/TestFileBasedSnapshotStore.java

```diff
@@ -0,0 +1,160 @@
+    return snapshotStore.purgePendingSnapshots();
+  }
+
+  @Override
+  public ActorFuture<Boolean> addSnapshotListener(final PersistedSnapshotListener listener) {
+    return snapshotStore.addSnapshotListener(listener);
+  }
+
+  @Override
+  public ActorFuture<Boolean> removeSnapshotListener(final PersistedSnapshotListener listener) {
+    return snapshotStore.removeSnapshotListener(listener);
+  }
+
+  @Override
+  public long getCurrentSnapshotIndex() {
+    return snapshotStore.getCurrentSnapshotIndex();
+  }
+
+  @Override
+  public ActorFuture<Void> delete() {
+    return snapshotStore.delete();
+  }
+
+  @Override
+  public Path getPath() {
+    return snapshotStore.getPath();
+  }
+
+  @Override
+  public ActorFuture<FileBasedReceivedSnapshot> newReceivedSnapshot(final String snapshotId) {
+    return snapshotStore.newReceivedSnapshot(snapshotId);
+  }
+
+  @Override
+  public void close() {
+    snapshotStore.close();
+  }
+
+  public void newSnapshot(final long index, final long term, final int size, final Random random) {
+    final var chunks =
+        IntStream.range(0, size)
+            .boxed()
+            .map(i -> "chunk-" + i)
+            .collect(Collectors.toMap(k -> k, v -> String.valueOf(random.nextLong())));
+    final var transientSnapshot =
+        snapshotStore.newTransientSnapshot(index, term, index, index).get();
+    transientSnapshot.take(p -> writeSnapshot(p, chunks)).join();
+    transientSnapshot.persist().join();
+  }
+
+  private boolean writeSnapshot(final Path path, final Map<String, String> chunks) {
+    try {
+      FileUtil.ensureDirectoryExists(path);
+
+      for (final var entry : chunks.entrySet()) {
+        final var fileName = path.resolve(entry.getKey());
+        final var fileContent = entry.getValue().getBytes(StandardCharsets.UTF_8);
+        Files.write(fileName, fileContent, CREATE_NEW, StandardOpenOption.WRITE);
+      }
+    } catch (final IOException e) {
+      throw new UncheckedIOException(e);
+    }
+    return true;
+  }
+
+  private static final class TestChecksumProvider implements CRC32CChecksumProvider {
+
+    @Override
+    public Map<String, Long> getSnapshotChecksums(final Path snapshotPath) {
+      final HashMap<String, Long> checksums = new HashMap<>();
+      try (final var files =
+          Files.newDirectoryStream(
+              snapshotPath, p -> p.getFileName().toString().startsWith("chunk"))) {
+        files.forEach(
+            file -> {
+              try {
+                final var fileContentChecksum = new CRC32C();
+                fileContentChecksum.update(Files.readAllBytes(file));
+                checksums.put(file.getFileName().toString(), fileContentChecksum.getValue());
+              } catch (final IOException e) {
+                throw new UncheckedIOException(e);
+              }
+            });
+      } catch (final IOException e) {
+        throw new UncheckedIOException(e);
+      }
+
+      return checksums;
+    }
+  }
```

Comment:
:thought_balloon: This shouldn't be needed, a no-op one that returns an empty map has the same behavior. A checksum provider is only useful if there are files for which we know the checksum _without_ reading all of the file contents somehow (i.e. because they are provided by RocksDB).

Reply:
A noop one do not reproduce the bug, because it results in calculating the combined checksum as it was before. In the tests, it makes sense to keep it as close to production usage to catch such inconsistencies.

Reply:
Oh right, I missed that aspect completely. All good then :+1: